openhands/agent_server/desktop_router.py

```diff
@@ -0,0 +1,47 @@
+"""Desktop router for agent server API endpoints."""
+
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel
+
+from openhands.agent_server.desktop_service import get_desktop_service
+from openhands.sdk.logger import get_logger
+
+
+logger = get_logger(__name__)
+
+desktop_router = APIRouter(prefix="/desktop", tags=["Desktop"])
+
+
+class DesktopUrlResponse(BaseModel):
+    """Response model for Desktop URL."""
+
+    url: str | None
+
+
+@desktop_router.get("/url", response_model=DesktopUrlResponse)
```

Comment:
I think we need to add `/api` to ensure this is authenticated

Reply:
Yep i think we do

It is part of the API Router
https://github.com/All-Hands-AI/agent-sdk/pull/515/files#diff-5d96de9766018f0ed11b03e7189c53cb2ea9f735a1f440f4817926d59bd605ebR113-R133